### ISSUE TITLE
Ctrl + backspace doesn't delete a whole word

### ISSUE THREAD

**Issue by ssougnez** [ckeditor/ckeditor5#414]
When hitting "ctrl + backspace", only one character is deleted while it should be the entire word.

**Comment by Reinmar** [comment]
Yep, we skipped that for a moment since it's not trivial to implement it considering all languages.

We have two options.

1. We can either try implementing it in JS and improve the implementation for less popular languages with time, based on bug reports.
2. Or we can try an approach where we don't use `preventDefault()` when discovering <kbd>Ctrl</kbd> + <kbd>Backspace</kbd> and let the browser delete a proper fragment of the text and try to work with mutations.

Actually, there's also a 3rd option – wait for the beforeinput event (or pick one of the above but use the event if it's present). beforeinput is a new event proposed by the [Editing TF (W3C)](http://w3c.github.io/editing/) and I've already [did some checks](https://github.com/w3c/editing/issues/149) for W3C how it works us. Back then, it allowed us to check which part of the text the browser would delete (so discovering words boundaries in the text would be done by the browser) but we were able to handle the deletion by ourselves (we were able to call `preventDefault()`). Blink and Webkit already have parts of this event working. Unfortunately, recently, Google changed their mind and it seems that we'll lose the ability to prevent the default action of the event, which means that we'll have to let the browser change the DOM and then we'll need to re-render the DOM as we see it, which may lead to some problems (mostly – with composition). Still, since we're able to quite reliably handle most of the input right now (using mutations and listening on backspace/delete keys), it may come with help to handle the other cases (deleting words, lines, etc).

PS. We'll also need to use some hybrid solution to handle backspacing on devices which don't provide key codes in `keydown` events – e.g. Chrome@Android. AFAIK (but I'd need to check again) there was no way to check which key was pressed, so we need other ways to figure this out.
